sdk/storage/azure-storage-blob/tests/test_container.py

```diff
@@ -1001,57 +1001,91 @@ def test_delete_blobs_simple(self):
-            blob1_client = container.get_blob_client('blob1')
-            blob1_client.upload_blob(data)
-            blob1_client.create_snapshot()
+            container.get_blob_client('blob1').upload_blob(data)
             container.get_blob_client('blob2').upload_blob(data)
             container.get_blob_client('blob3').upload_blob(data)
         except:
             pass
-        blobs = list(container.list_blobs(include='snapshots'))
-        assert len(blobs) == 4  # 3 blobs + 1 snapshot
 
         # Act
         response = container.delete_blobs(
             'blob1',
             'blob2',
             'blob3',
-            delete_snapshots='only'
+            raise_on_any_failure=False
         )
+        response = list(response)
         assert len(response) == 3
         assert response[0].status_code == 202
-        assert response[1].status_code == 404  # There was no snapshot
-        assert response[2].status_code == 404  # There was no snapshot
+        assert response[1].status_code == 202
+        assert response[2].status_code == 202
+
+    @pytest.mark.skipif(sys.version_info < (3, 0), reason="Batch not supported on Python 2.7")
+    @record
+    def test_delete_blobs_snapshot(self):
+        # Arrange
+        container = self._create_container()
+        data = b'hello world'
 
+        try:
+            blob1_client = container.get_blob_client('blob1')
+            blob1_client.upload_blob(data)
+            blob1_client.create_snapshot()
+            container.get_blob_client('blob2').upload_blob(data)
+            container.get_blob_client('blob3').upload_blob(data)
+        except:
+            pass
         blobs = list(container.list_blobs(include='snapshots'))
-        assert len(blobs) == 3  # 3 blobs
+        assert len(blobs) == 4  # 3 blobs + 1 snapshot
+
+        # Act
+        try:
+            response = container.delete_blobs(
+                'blob1',
+                'blob2',
+                'blob3',
+                delete_snapshots='only'
+            )
+        except HttpResponseError as err:
+            parts = list(err.parts)
+            assert len(parts) == 3
+            assert parts[0].status_code == 202
+            assert parts[1].status_code == 404  # There was no snapshot
+            assert parts[2].status_code == 404  # There was no snapshot
+
+            blobs = list(container.list_blobs(include='snapshots'))
+            assert len(blobs) == 3  # 3 blobs
 
     @pytest.mark.skipif(sys.version_info < (3, 0), reason="Batch not supported on Python 2.7")
     @record
     def test_standard_blob_tier_set_tier_api_batch(self):
         container = self._create_container()
         tiers = [StandardBlobTier.Archive, StandardBlobTier.Cool, StandardBlobTier.Hot]
 
-        response = container.delete_blobs(
-            'blob1',
-            'blob2',
-            'blob3',
-        )
-
         for tier in tiers:
+            try:
+                response = container.delete_blobs(
```

Comment:
Being this is not the point of this test, I will just pass "raise_on_first=False" instead